src/explore-education-statistics-common/src/modules/table-tool/components/TableHeadersForm.tsx

```diff
@@ -50,113 +56,202 @@ const TableHeadersForm = ({
     [onSubmit],
   );
 
+  const handleMoveGroupToOtherAxis = useMemo(
+    () => (
+      sourceId: keyof FormValues,
+      groupIndex: number,
+      form: FormikProps<FormValues>,
+    ) => {
+      const destinationId: keyof FormValues =
+        sourceId === 'rowGroups' ? 'columnGroups' : 'rowGroups';
+
+      const nextSourceValue = [...form.values[sourceId]];
+      const sourceItem = nextSourceValue.splice(groupIndex, 1);
+      const nextDestinationValue = [...form.values[destinationId]].concat(
+        sourceItem,
+      );
+
+      form.setFieldValue(sourceId, nextSourceValue);
+      form.setFieldValue(destinationId, nextDestinationValue);
+      form.setFieldTouched(sourceId);
+      form.setFieldTouched(destinationId);
```

Comment:
We're doing a bunch of the same stuff in the DragDropContext's `onDragEnd`. Perhaps we could refactor this out into a function and re-use it in both places.